tiny_dnn/layers/max_unpooling_layer.h

```diff
@@ -88,7 +88,9 @@ class max_unpooling_layer : public layer {
       for_(parallelize_, 0, in2out_.size(), [&](const blocked_range &r) {
         for (int i = r.begin(); i < r.end(); i++) {
           const auto &in_index = out2in_[i];
-          out_vec[i] = (max_idx[in_index] == i) ? in_vec[in_index] : float_t{0};
+          out_vec[i]           = (static_cast<int>(max_idx[in_index]) == i)
```

Comment:
Oh, right, I misread it.
Does it make sense to make `blocked_range` elements of type `serial_size_t` then (not really sure how this part works)?